src/main/java/teammates/logic/api/GateKeeper.java

```diff
@@ -324,24 +324,34 @@ private boolean isStudent() {
     }
 
     public void verifyAccessibleForCurrentUserAsInstructorOrTeamMember(AccountAttributes account, String courseId,
-            String section, String email) {
+                                                                       String section, String email) {
+        if (!isValidInstructor(section, courseId, account) && !isValidStudent(email, courseId, account)) {
+            throw new UnauthorizedAccessException("User is not in the course that student belongs to");
```

Comment:
If we end up implementing all the changes I have suggested above, then we might have to change the error message to `"User does not have enough privileges to view the photo"`.